File: app/src/main/java/com/example/lanhuajian/blues/module_login/AccountLoginFragment.java
package com.example.lanhuajian.blues.module_login;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import com.example.lanhuajian.blues.MainActivity;
import com.example.lanhuajian.blues.R;
import com.example.lanhuajian.blues.framework.base.BaseFragment;
import com.example.lanhuajian.blues.framework.utils.HelperUtil;

public class AccountLoginFragment extends BaseFragment implements View.OnClickListener, TextWatcher, LoginContract.iLoginContractView {

    private EditText etAccount, etPassword;
    private TextView tvLogin, tvSkip;

    private String account, password;

    private LoginContract.iLoginContractPresenter iPresenter;


    @Override
    public int setLayoutResourceId() {
        return R.layout.fragment_account_login;
    }

    @Override
    public void initLayout(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        etAccount = rootView.findViewById(R.id.et_input_account);
        etPassword = rootView.findViewById(R.id.et_input_password);

        tvSkip = rootView.findViewById(R.id.tv_skip);
        tvLogin = rootView.findViewById(R.id.tv_login);

        mPresenter = new LoginPresenter(this);
    }

    @Override
    public void lazyFetchData() {

    }

    @Override
    public void setListener() {
        etAccount.addTextChangedListener(this);
        etPassword.addTextChangedListener(this);

        tvLogin.setOnClickListener(this);
        tvSkip.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tv_skip:
//                getActivity().finish();
                break;
            case R.id.tv_login:
                //调用登陆接口
                iPresenter.doLogin(account, password);
                break;
        }
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        account = etAccount.getText().toString().trim();
        password = etPassword.getText().toString().trim();

    }

    @Override
    public void afterTextChanged(Editable s) {

    }


    @Override
    public void onSuccess(LoginResponse result) {
        HelperUtil.showToastShort("登陆成功");
        //登陆成功进行跳转
        getmContext().startActivity(new Intent(getmContext(), MainActivity.class));
        //跳转结束 finish上一个activity
        getmContext().finish();
    }

    @Override
    public void onFailure(String result) {
        //登陆失败进行提示
        HelperUtil.showToastShort(result);
    }

    @Override
    public void setPresenter(LoginContract.iLoginContractPresenter presenter) {
        iPresenter = presenter;
    }

    @Override
    public void showBegin() {

    }

    @Override
    public void showLoading() {

    }

    @Override
    public void showFinished() {

    }

    @Override
    public void showError() {

    }

    @Override
    public void showEmpty() {

    }
}

File: app/src/main/java/com/blues/module_login/view/AccountLoginFragment.kt
package com.blues.module_login.view

import android.content.Intent
import com.blues.framework.base.BaseKoinFragment
import android.widget.EditText
import android.widget.TextView
import com.blues.R
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import com.blues.MainActivity
import com.blues.adapter.TextWatcherAdapter
import com.blues.framework.utils.HelperUtil
import com.blues.module_login.vm.LoginViewModel
import org.koin.androidx.viewmodel.ext.android.viewModel

class AccountLoginFragment : BaseKoinFragment(), View.OnClickListener {

    private lateinit var etAccount: EditText
    private lateinit var etPassword: EditText
    private lateinit var tvLogin: TextView
    private lateinit var tvSkip: TextView
    private var account: String? = null
    private var password: String? = null

    private val loginViewModel: LoginViewModel by viewModel()

    fun lazyFetchData() {}
    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_skip -> {
                startActivity(Intent(requireActivity(), MainActivity::class.java))
            }
            R.id.tv_login -> {
                if (account == null) {
                    HelperUtil.showToast("账号不能为空!!!")
                }

                if (password == null) {
                    HelperUtil.showToast("密码不能为空!!!")
                }

                //调用登陆接口
                loginViewModel.login(account!!, password!!)
            }
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.fragment_account_login
    }

    override fun initData(inflater: LayoutInflater, container: ViewGroup?, saveInstanced: Bundle?) {
        rootView?.let {
            etAccount = it.findViewById<EditText>(R.id.et_input_account)
                .apply {
                    addTextChangedListener(object : TextWatcherAdapter() {
                        override fun onTextChanged(s: CharSequence?, start: Int, before: Int,
                            count: Int) {
                            account = text.toString()
                                .trim()
                        }
                    })
                }
            etPassword = it.findViewById<EditText>(R.id.et_input_password)
                .apply {
                    addTextChangedListener(object : TextWatcherAdapter() {
                        override fun onTextChanged(s: CharSequence?, start: Int, before: Int,
                            count: Int) {
                            password = text.toString()
                                .trim()
                        }
                    })
                }
            tvSkip = it.findViewById<TextView>(R.id.tv_skip)
                .apply {
                    setOnClickListener(this@AccountLoginFragment)
                }
            tvLogin = it.findViewById<TextView>(R.id.tv_login)
                .apply {
                    setOnClickListener(this@AccountLoginFragment)
                }
        }
    }

    override fun observe() {
        loginViewModel.result.observe(this) {
            if (it.errorCode == 200) {
                startActivity(Intent(requireActivity(), MainActivity::class.java))
            } else {
                HelperUtil.showToast(it.errorMsg)
            }
        }
    }
}